sdk/src/test/scala/io/horizen/account/state/ContractInteropStackTest.scala

```diff
@@ -0,0 +1,125 @@
+package io.horizen.account.state
+
+import io.horizen.evm.{Address, TraceOptions, Tracer}
+import io.horizen.utils.BytesUtils
+import org.junit.Assert.assertEquals
+import org.junit.{Ignore, Test}
+import sparkz.crypto.hash.Keccak256
+
+import java.nio.ByteBuffer
+import scala.util.Try
+
+class ContractInteropStackTest extends ContractInteropTestBase {
+
+  override val processorToTest: NativeSmartContractMsgProcessor = NativeTestContract
+
+  /**
+   * Native Contract to test correct call depth limiting: The input is parsed into three arguments
+   *   - function signature: ignored here, but useful in combination with an EVM-based contract
+   *   - target: address of the next contract to call
+   *   - counter: to keep track of maximum reached call depth
+   *
+   * Execution causes one nested call to the given target address with the input set to:
+   *   - function signature: passed through unchanged
+   *   - target: the contracts own address
+   *   - counter: incremented and passed along
+   *
+   * The return value is:
+   *   - The result of the nested call in case of success.
+   *   - The current counter if the nested call failed.
+   *
+   * Note this enables multiple test cases:
+   *   - Passing the contract address itself as target causes an infinite recursive loop.
+   *   - Passing a different target address allows that contract to call back to this one to cause a loop between two
+   *     contracts.
+   */
+  private object NativeTestContract extends NativeSmartContractMsgProcessor {
+    override val contractAddress: Address = new Address("0x00000000000000000000000000000000deadbeef")
+    override val contractCode: Array[Byte] = Keccak256.hash("NativeInteropStackContract")
+
+    override def process(
+        invocation: Invocation,
+        view: BaseAccountStateView,
+        context: ExecutionContext
+    ): Array[Byte] = {
+      // parse input
+      val in = invocation.input
+//      println(s"call at depth ${context.depth - 1} with ${invocation.gasPool.getGas} gas and input (${in.length}): ${BytesUtils.toHexString(in)}")
+      if (in.length != 4 + 32 + 32) {
+        throw new IllegalArgumentException("NativeInteropStackContract called with invalid arguments")
+      }
+//      val signature = in.take(4)
```

Comment:
Can this and other println commented statements be removed?

Reply:
Done